File: WinFormsApp1/Form1.vb
Imports System.Globalization
Public Class Form1
    Private ReadOnly _lastChecked As New LastChecked() With {.Index = -1}
    Private Sub Form1_Shown(sender As Object, e As EventArgs) Handles Me.Shown
        Dim months = DateTimeFormatInfo.CurrentInfo.MonthNames.Take(12).ToList()
        checkedListBox1.DataSource = months
    End Sub

    Private Sub checkedListBox1_ItemCheck(sender As Object, e As ItemCheckEventArgs) Handles checkedListBox1.ItemCheck
        _lastChecked.CheckState = e.NewValue
        _lastChecked.Index = e.Index
    End Sub

    Private Sub LastCheckedButton_Click(sender As Object, e As EventArgs) Handles LastCheckedButton.Click
        If _lastChecked.Index > -1 Then
            MessageBox.Show($"Index: {_lastChecked.Index,-5}{_lastChecked.CheckState}")
        Else
            MessageBox.Show("Nothing checked yet")
        End If
    End Sub
End Class

Public Class LastChecked
    Public Property Index() As Integer
    Public Property CheckState() As CheckState
End Class

File: CheckedListBoxLastChecked/Form1.cs
using System;
using System.Globalization;
using System.Linq;
using System.Windows.Forms;

namespace CheckedListBoxLastChecked
{
    public partial class Form1 : Form
    {
        private readonly LastChecked _lastChecked = new LastChecked() { Index = -1 };
        public Form1()
        {
            InitializeComponent();
            Shown += OnShown;
        }

        private void OnShown(object sender, EventArgs e)
        {
            var months = DateTimeFormatInfo.CurrentInfo.MonthNames.Take(12).ToList();
            checkedListBox1.DataSource = months;
            checkedListBox1.ItemCheck += CheckedListBox1OnItemCheck;
        }

        private void CheckedListBox1OnItemCheck(object sender, ItemCheckEventArgs e)
        {

            _lastChecked.CheckState = e.NewValue;
            _lastChecked.Index = e.Index;

        }

        private void LastCheckedButton_Click(object sender, EventArgs e)
        {
            if (_lastChecked.Index > -1)
            {
                //MessageBox.Show($"Index: {_lastChecked.Index,-5}{_lastChecked.CheckState}");
                Console.WriteLine(checkedListBox1.Items[_lastChecked.Index]);
                if (_lastChecked.CheckState == CheckState.Checked)
                {

                }
                else
                {

                }
            }
            else
            {
                MessageBox.Show("Nothing checked yet");
            }
        }

    }
    // should be in its own file
    public class LastChecked
    {
        public int Index { get; set; }
        public CheckState CheckState { get; set; }

    }
}
